components/sidebar/sidebar.jsx

```diff
@@ -0,0 +1,692 @@
+            </OverlayTrigger>
+        );
+
+        if (!ChannelUtils.showCreateOption(Constants.OPEN_CHANNEL, this.props.isTeamAdmin, this.props.isSystemAdmin)) {
+            createPublicChannelIcon = null;
+        }
+
+        const createDirectMessageIcon = (
+            <OverlayTrigger
+                className='hidden-xs'
+                delayShow={500}
+                placement='top'
+                overlay={createDirectMessageTooltip}
+            >
+                <button
+                    className='add-channel-btn cursor--pointer style--none'
+                    onClick={this.handleOpenMoreDirectChannelsModal}
+                >
+                    {'+'}
+                </button>
+            </OverlayTrigger>
+        );
+
+        if (!ChannelUtils.showCreateOption(Constants.PRIVATE_CHANNEL, this.props.isTeamAdmin, this.props.isSystemAdmin)) {
+            createPrivateChannelIcon = null;
+        }
+
+        let moreDirectChannelsModal;
+        if (this.state.showDirectChannelsModal) {
+            moreDirectChannelsModal = (
+                <MoreDirectChannels
+                    onModalDismissed={this.hideMoreDirectChannelsModal}
+                    startingUsers={this.state.startingUsers}
+                />
+            );
+        }
+
+        let moreChannelsModal;
+        if (this.state.showMoreChannelsModal) {
+            moreChannelsModal = (
+                <MoreChannels
+                    onModalDismissed={this.hideMoreChannelsModal}
+                    handleNewChannel={() => {
+                        this.hideMoreChannelsModal();
+                        this.showNewChannelModal(Constants.OPEN_CHANNEL);
+                    }}
+                />
+            );
+        }
+
+        let quickSwitchTextShortcutId = 'quick_switch_modal.channelsShortcut.windows';
+        let quickSwitchTextShortcutDefault = '- CTRL+K';
+        if (Utils.isMac()) {
+            quickSwitchTextShortcutId = 'quick_switch_modal.channelsShortcut.mac';
+            quickSwitchTextShortcutDefault = '- ⌘K';
+        }
+
+        const quickSwitchTextShortcut = (
+            <span className='switch__shortcut hidden-xs'>
+                <FormattedMessage
+                    id={quickSwitchTextShortcutId}
+                    defaultMessage={quickSwitchTextShortcutDefault}
+                />
+            </span>
+        );
+
+        return (
+            <div
+                className='sidebar--left'
+                id='sidebar-left'
+                key='sidebar-left'
+            >
+                <NewChannelFlow
+                    show={showChannelModal}
+                    channelType={this.state.newChannelModalType}
+                    onModalDismissed={this.hideNewChannelModal}
+                />
+                {moreDirectChannelsModal}
+                {moreChannelsModal}
+
+                <SidebarHeader
+                    teamDisplayName={this.props.currentTeam.display_name}
```

Comment:
I like the idea, but It can be done in other PR, probably in a SidebarHeader migration to redux/pure-component. I think this PR is quite big already, so I prefer to not keep adding complexity.